obsplus/events/validate.py

```diff
@@ -100,57 +99,191 @@ def check_origins(event: Event):
 
 
 # register the nullish nslc code replacement
-catalog_validator(replace_null_nlsc_codes)
+validator("obsplus", Event)(replace_null_nlsc_codes)
 
 
-def validate_catalog(events: Union[Catalog, Event],) -> Optional[Union[Catalog, Event]]:
+@validator("obsplus", Event)
+def check_duplicate_picks(event: Event, **kwargs):
+    """
+    Ensure there are no picks with the same phases on the same channels.
+    """
+    # first get dataframe of picks
+    pdf = obsplus.picks_to_df(event)
+    pdf = pdf.loc[pdf.evaluation_status != "rejected"]
+    event_id = str(event.resource_id)
+
+    def dup_picks(phase_hint, on="station"):
+        """ function for checking """
+        df = pdf.loc[pdf.phase_hint == phase_hint]
+        bad = df.loc[df[on].duplicated()][on].tolist()
+        assert len(bad) == 0, (
+            f"Duplicate {phase_hint} picks found\n"
+            f"event_id: {event_id}, "
+            f"{on}/s: {bad}"
+        )
+
+    # Checking for duplicated picks
+    dup_picks("P")
```

Comment:
This covers the basics, but I wonder if it would be simple to set this up to work with any arbitrary `phase_hint`s?